hummingbird/ml/operator_converters/sparkml/sql_transformer.py

```diff
@@ -0,0 +1,116 @@
+
+
+class SQLTransformerModel(BaseOperator, torch.nn.Module):
+
+    # List of supported binary SQL ops.
+    supported_binary_ops = {
+        'org.apache.spark.sql.catalyst.expressions.Add': torch.add,
+        'org.apache.spark.sql.catalyst.expressions.Subtract': torch.sub,
+        'org.apache.spark.sql.catalyst.expressions.Multiply': torch.mul,
+        'org.apache.spark.sql.catalyst.expressions.Divide': torch.div,
+    }
+
+    def __init__(self, input_names, project_list, device):
+        super(SQLTransformerModel, self).__init__()
+        self.transformer = True
+        self.input_names = input_names
+        self.project_list = project_list
+
+    def forward(self, *x):
+        inputs_dict = {name : tensor for name, tensor in zip(self.input_names, x)}
+        # The first node of the project_node_def is the name definition node. Hence ignored.
+        return [self.calculate_output_feature(project_node_def_stack[1:], inputs_dict)[0] for project_node_def_stack in self.project_list]
+
+    def calculate_output_feature(self, project_node_def_stack, inputs_dict):
+        if project_node_def_stack[0]['class'] == 'org.apache.spark.sql.catalyst.analysis.UnresolvedAttribute':
+            return inputs_dict[project_node_def_stack[0]['nameParts'][1:-1].strip()], project_node_def_stack[1:]
+        elif project_node_def_stack[0]['class'] == 'org.apache.spark.sql.catalyst.expressions.Literal':
+            return float(project_node_def_stack[0]['value']), project_node_def_stack[1:]
+        elif project_node_def_stack[0]['class'] in self.supported_binary_ops:
+            op = self.supported_binary_ops[project_node_def_stack[0]['class']]
+            project_node_def_stack = project_node_def_stack[1:]
+            left, project_node_def_stack = self.calculate_output_feature(project_node_def_stack, inputs_dict)
+            right, project_node_def_stack = self.calculate_output_feature(project_node_def_stack, inputs_dict)
+            return op(left, right), project_node_def_stack
+        else:
+            raise RuntimeError('SQLTransformer encountered unsupported operator type: {}'.format(project_node_def_stack[0]['class']))
+
+
+def convert_sparkml_sql_transformer(operator, device, extra_config):
+    """
+    Converter for `pyspark.ml.feature.SQLTransformer`
+
+    Args:
+        operator: An operator wrapping a `pyspark.ml.feature.SQLTransformer`
+        device: String defining the type of device the converted operator should be run on
+        extra_config: Extra configuration used to select the best conversion strategy
+
+    Returns:
+        A PyTorch model
+    """
+    from pyspark.sql import SparkSession
+
+    spark = SparkSession.builder.getOrCreate()
+    parser = spark._jsparkSession.sessionState().sqlParser()
+    statement = operator.raw_operator.getStatement()
+    plan = parser.parsePlan(statement)
```

Comment:
This is "just" a parsed plan, right? Meaning the query it is not actually optimized. In general I think it is ok, but bonus point if we can somehow get the optimized logical plan. I think this will also solve the problems with `unresolvedAttribute`, `stars`, etc.